core-api/src/test/java/com/optimizely/ab/config/audience/AudienceConditionEvaluationTest.java

```diff
@@ -108,6 +114,65 @@ public void invalidMatchCondition() throws Exception {
     public void invalidMatch() throws Exception {
         UserAttribute testInstance = new UserAttribute("browser_type", "custom_attribute", "blah", "chrome");
         assertNull(testInstance.evaluate(null, testUserAttributes));
+        logbackVerifier.expectMessage(Level.WARN, String.format("Audience condition \"%s\" uses an unknown match type. You may need to upgrade to a newer release of the Optimizely SDK", testInstance.toString()));
+    }
+
+    /**
+     * Verify that UserAttribute.evaluate returns null on invalid attribute type.
+     */
+    @Test
+    public void unexpectedAttributeType() throws Exception {
+        UserAttribute testInstance = new UserAttribute("browser_type", "custom_attribute", "gt", 20);
+        assertNull(testInstance.evaluate(null, testUserAttributes));
+        logbackVerifier.expectMessage(Level.ERROR, "Greater than match failed");
+        logbackVerifier.expectMessage(Level.WARN, String.format("Audience condition \"%s\" evaluated to UNKNOWN because a value of type \"java.lang.String\" was passed for user attribute \"browser_type\"", testInstance.toString()));
+    }
+
+    /**
+     * Verify that UserAttribute.evaluate returns null on invalid attribute type.
+     */
+    @Test
+    public void unexpectedAttributeTypeNull() throws Exception {
+        UserAttribute testInstance = new UserAttribute("browser_type", "custom_attribute", "gt", 20);
+        assertNull(testInstance.evaluate(null, Collections.singletonMap("browser_type", null)));
+        logbackVerifier.expectMessage(Level.ERROR, String.format("Cannot evaluate targeting condition since the value for attribute is an incompatible type"));
+        logbackVerifier.expectMessage(Level.ERROR, String.format("Greater than match failed"));
```

Comment:
Ah, I know that these errors were already being logged before the current PR, but we don't want them any longer since we only want warnings instead. Can you remove the error logging from AttributeMatch.java, SubstringMatch.java, ExactNumberMatch.java, GTMatch.java, and LTMatch.java?

Thanks for including these assertion in the initial tests, however! I wouldn't have noticed otherwise. 😬